config/config.go

```diff
@@ -560,6 +560,11 @@ func FindConfigFilesInPath(rootPath string, terragruntOptions *options.Terragrun
 		}
 
 		if isTerragruntModule {
+			// Skip Module if search file is not found in layer
+			if terragruntOptions.LayerHasFile != "" && !util.FileExists(path+terragruntOptions.LayerHasFile) {
```

Comment:
Should use the auxiliary functions for joining filepaths to avoid unexpected problems. 

```suggestion
			if terragruntOptions.LayerHasFile != "" && !util.FileExists(util.JoinPath(path, terragruntOptions.LayerHasFile)) {
```